Tests/TestMapBase.gd

```diff
@@ -0,0 +1,20 @@
+class_name TestMapBase extends GutTest
+
+var test_generator: Script = load("res://Map/MapManager.gd")
+var map_manager: MapManager = null
+
+var test_width: Array[int] = [1,3,5,3,1]
+
+var test_map: MapBase = null
+
+func before_each() -> void:
+	map_manager = test_generator.new()
+	MapManager.map_width_array = test_width 
+	# need to change the manager since it's globally loaded
+	test_map = map_manager.create_map(test_width)
```

Comment:
Why is `test_width` being passed to `MapManager `and then used as an argument in `map_manager.create_map` ?

Reply:
why not ?
if you look at MapManager, it has a `map_width_array` argument, so I give it the test width
then I need the map so use the `ŧest_width` again
I'm basically doing the same thing as the `_ready`

Reply:
What I'm saying is that `create_map `has a redundant argument.
`map_width_array `already stores what the argument, `map_floors_width`, is passing to the function.
I can understand it if it serves the purpose of making `create_map ` just use local variables

Reply:
Here is the code of `create_map`
```GDScript
func create_map(map_floors_width: Array[int]) -> MapBase: ## Generates and Populates a map with rooms that have random room types. More in depth algorithms will be added in the future
	var _map: MapBase = MapBase.new()
	var _grid: Array[Array] = [] ## 2d array to return. this will be populated with rooms
	var _max_floor_size: int = map_floors_width.max()
	for index_height: int in range(map_floors_width.size()): ## loop through the floor and append null to create a square grid
		
		var _floor_width : int = map_floors_width[index_height]
		var _padding_size : int = (_max_floor_size - _floor_width)/2
		
		var _padding:Array = []
		_padding.resize(_padding_size)
		_padding.fill(null)
		_grid.append([])
		_grid[index_height].append_array(_padding)
		
		for index_width: int in range(map_floors_width[index_height]):## loop through elements of the grid and assign a room type
			var _rand_type_index: int = randi_range(0, GlobalVar.EVENTS_CLASSIFICATION.size() - 1)
			var _room_event: EventBase = GlobalVar.EVENTS_CLASSIFICATION[_rand_type_index].new()
			var _generated_room: RoomBase = RoomBase.new()
			_generated_room.room_event = _room_event
			_generated_room.room_position = Vector2i(index_width + _padding_size, index_height)
			_grid[index_height].append(_generated_room as RoomBase)
		_grid[index_height].append_array(_padding)
	_map.rooms = _grid
	return _map as MapBase
```

`create_map` doesn't have redundant arguments, you *need* to give it an array of width
I could write that instead:
```GDScript
MapManager.map_width_array = test_width 
# need to change the manager since it's globally loaded
test_map = map_manager.create_map(MapManager.map_width_array)
```
But it's longer to write and does exactly the same thing